src/Project.cpp

```diff
@@ -130,8 +134,20 @@ auto Project::getWorld(const IID& iid) const -> const World& {
     ldtk_error("World with IID \""+iid.str()+"\" not found in Project \""+getFilePath().c_str()+"\".");
 }
 
-void Project::load(const nlohmann::json& j) {
-    m_default_pivot.x = j["defaultPivotX"].get<float>();
+auto ldtk::Project::defaultFileLoader(const std::string& filepath) -> std::string {
+    std::ifstream in(filepath);
+    if (in.fail()) {
+        ldtk_error("Failed to open file \"" + filepath + "\" : " + strerror(errno));
+    }
+
+    std::ostringstream file_string_stream;
+    file_string_stream << in.rdbuf();
+
+    return file_string_stream.str();
```

Comment:
I wonder what are the performance impact of putting the whole file in a string.

I think this could impact large project files, I think there should be a better approach.

Reply:
This was something on my mind too. I'll see if I can remove the `defaultFileLoader` entirely and just load it from the file in situ in the overload without a file loader. I'll look hereinto later this weekend.

Another idea would be for the file loader function to return `nlohmann::json` instead of `std::string`, but that means the JSON library would have to be exposed fully to the user, so I don't think it will be very practical in that regard.

I'm open to other suggestions.